src/xref2/link.ml

```diff
@@ -365,6 +365,51 @@ and open_ env parent = function
   | { Odoc_model__Lang.Open.doc; _ } as open_ ->
       { open_ with doc = comment_docs env parent doc }
 
+let warn_on_hidden_representation (id : Id.Type.t)
+    (r : Lang.TypeDecl.Representation.t) =
+  let open Paths.Identifier in
+  let rec read_typ_exp typ_expr =
```

Comment:
`typ_contains_hidden` would be a better name ?

Reply:
True! I just moved the code from generator.ml and didn't think enough about the name. I'll fix this, good call.